gate-web/gate-web.gradle

```diff
@@ -43,13 +42,18 @@ dependencies {
 
   compileOnly spinnaker.dependency("lombok")
 
+  testCompile project(":gate-ldap") // TODO: Move system tests to own module
   testCompile "com.squareup.okhttp:mockwebserver:${spinnaker.version('okHttp')}"
   testCompile apachedsDependencies
   testCompile spinnaker.dependency("korkJedisTest")
 
   //this brings in the jetty GzipFilter which boot will autoconfigure
   runtime 'org.eclipse.jetty:jetty-servlets:9.2.11.v20150529'
-  runtime project(":gate-auth-iap")
+  runtime project(":gate-iap")
```

Comment:
What benefit is declaring these as `runtime` dependencies over `compile` ?

Reply:
Forces `gate-web` to not directly import classes from any of the providers and instead rely on Spring component scans. Not absolutely required in this PR but will be needed in the next one that allows building only selected providers